src/main/scala/DataMunging.scala

```diff
@@ -0,0 +1,40 @@
+package DataMunging
+import scala.io.Source
+
+object DataMunging {
+
+    def main(args: Array[String]) ={
+            getMinSpread()
+            getMinGoalDiff()
+    }
+    def getMinSpread() = {
+        val source = Source.fromFile("weather.dat")
```

Comment:
You could also factor out
1.  the code to fetch all the lines in the function, because that is common to both getMinSpread() and getMinGoalDiff(). 
2.  filter out the header and the lines with "-----" and the blank line in this function. 
3. and you tokenise (split) by space.
That is: 

``` 
def getLines(val fileName:String):List[Array[String]] = {
               val source = Source.fromFile(fileName)
               val lines = source.getLines.toList
                source.close
               lines.drop(0)
                       .filter(line=> !line.contains("-----"))
                       .filter(line => !line.trim.isEmpty)
                       .map(_.split("\\s+"))
   }
```